packages/firestore/src/local/memory_persistence.ts

```diff
@@ -57,24 +57,30 @@ export class MemoryPersistence implements Persistence {
    * will make the in-memory persistence layer behave as if it were actually
    * persisting values.
    */
-  private mutationQueues: { [user: string]: MutationQueue } = {};
+  private mutationQueues: { [user: string]: MemoryMutationQueue } = {};
   private remoteDocumentCache = new MemoryRemoteDocumentCache();
   private readonly queryCache: MemoryQueryCache; // = new MemoryQueryCache();
   private readonly listenSequence = new ListenSequence(0);
 
   private _started = false;
 
-  // TODO(gsoltis): remove option to be null once eager delegate is implemented.
-  private _referenceDelegate: ReferenceDelegate | null;
+  readonly referenceDelegate: MemoryLruDelegate | MemoryEagerDelegate;
 
   static createLruPersistence(clientId: ClientId): MemoryPersistence {
-    const persistence = new MemoryPersistence(clientId);
-    persistence._referenceDelegate = new MemoryLruDelegate(persistence);
-    return persistence;
+    return new MemoryPersistence(clientId, /* isEager= */ false);
```

Comment:
What do you think about passing  in the `referenceDelegate` and removing the `isEager` flag?

Reply:
Sounds good. Thanks for the explanation.